core/src/main/java/org/elasticsearch/index/query/TermsQueryParser.java

```diff
@@ -101,13 +80,14 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
                     throw new QueryParsingException(parseContext, "[terms] query does not support multiple fields");
                 }
                 fieldName = currentFieldName;
-
+                
+                values = Lists.newArrayList();
```

Comment:
I didn't notice that we initialize values straight-away, that might change though, see https://github.com/elastic/elasticsearch/pull/12042/files#r37095081